src/PSDMatrices.jl

```diff
@@ -1,13 +1,64 @@
+
 module PSDMatrices
 
-using LinearAlgebra
-import LinearAlgebra: cholesky, diag, det, logdet
+import Base: \, /, size, inv, copy, ==
+import LinearAlgebra: cholesky, det, logdet, qr, triu, diag, UpperTriangular
+
+struct PSDMatrix{T,Ltype} <: AbstractMatrix{T}
+    L::Ltype
+    PSDMatrix(L::AbstractMatrix{T}) where {T} = new{T,typeof(L)}(L)
+end
+
+# Custom functions
```

Comment:
Should we move this section to the bottom of the file? That way it is clear that the type implements the AbstractArray interface, before we add our custom functionality.

Reply:
How about I do this in the following PR?

Reply:
But let's do that at some point soon, since moving code always gives larger diffs ;)

Reply:
okay you're right. I moved it.